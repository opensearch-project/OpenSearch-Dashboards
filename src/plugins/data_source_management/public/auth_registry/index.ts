/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export {
  IAuthenticationMethodRegistry,
  AuthenticationMethod,
  AuthenticationMethodRegistry,
} from './authentication_methods_registry';
