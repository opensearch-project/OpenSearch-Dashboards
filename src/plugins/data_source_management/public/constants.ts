/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { DirectQueryDatasourceType } from './types';

export const QUERY_RESTRICTED = 'query-restricted';
export const QUERY_ALL = 'query-all';

export const DatasourceTypeToDisplayName: { [key in DirectQueryDatasourceType]: string } = {
  PROMETHEUS: 'Prometheus',
  S3GLUE: 'Amazon S3',
};

export const PROMETHEUS_URL = 'Prometheus';
export const AMAZON_S3_URL = 'AmazonS3AWSGlue';
export const OPENSEARCH_URL = 'OpenSearch';

export const UrlToDatasourceType: { [key: string]: DirectQueryDatasourceType } = {
  [PROMETHEUS_URL]: 'PROMETHEUS',
  [AMAZON_S3_URL]: 'S3GLUE',
};
