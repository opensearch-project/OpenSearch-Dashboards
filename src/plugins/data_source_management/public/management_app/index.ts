/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { mountManagementSection } from './mount_management_section';
