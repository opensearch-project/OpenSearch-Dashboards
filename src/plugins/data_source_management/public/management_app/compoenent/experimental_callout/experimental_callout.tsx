/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiCallOut, EuiLink, EuiSpacer } from '@elastic/eui';
import { DocLinksStart } from 'opensearch-dashboards/public';
import {
  EXPERIMENTAL_FEATURE,
  EXPERIMENTAL_FEATURE_CALL_OUT_DESCRIPTION,
  DATA_SOURCE_LEAVE_FEEDBACK_TEXT,
  DATA_SOURCE_DOCUMENTATION_TEXT,
  DATA_SOURCE_OPEN_FORUM_TEXT,
} from '../../../components/text_content';

export const ExperimentalCallout = ({ docLinks }: { docLinks: DocLinksStart }) => {
  return (
    <>
      <EuiCallOut
        title={EXPERIMENTAL_FEATURE}
        iconType="iInCircle"
        data-test-subj="data-source-experimental-call"
      >
        <p>
          {EXPERIMENTAL_FEATURE_CALL_OUT_DESCRIPTION}
          <EuiLink href={docLinks.links.noDocumentation.indexPatterns.introduction} target="_blank">
            {DATA_SOURCE_DOCUMENTATION_TEXT}
          </EuiLink>{' '}
          {DATA_SOURCE_LEAVE_FEEDBACK_TEXT}
          <EuiLink href={docLinks.links.noDocumentation.openSearchForum} target="_blank">
            {DATA_SOURCE_OPEN_FORUM_TEXT}
          </EuiLink>
        </p>
      </EuiCallOut>
      <EuiSpacer size="m" />
    </>
  );
};
