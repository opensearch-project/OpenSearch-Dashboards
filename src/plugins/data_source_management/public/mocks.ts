/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { throwError } from 'rxjs';
import { SavedObjectsClientContract } from 'opensearch-dashboards/public';
import { AuthType } from './types';
import { coreMock } from '../../../core/public/mocks';

/* Mock Types */

export const docLinks = {
  links: {
    noDocumentation: {
      indexPatterns: {
        introduction: '',
      },
      scriptedFields: {},
    },
  },
};

const createDataSourceManagementContext = () => {
  const {
    chrome,
    application,
    savedObjects,
    uiSettings,
    notifications,
    overlays,
  } = coreMock.createStart();
  const { http } = coreMock.createSetup();

  return {
    chrome,
    application,
    savedObjects,
    uiSettings,
    notifications,
    overlays,
    http,
    docLinks,
    setBreadcrumbs: () => {},
  };
};

export const mockManagementPlugin = {
  createDataSourceManagementContext,
  docLinks,
};

/* Mock data responses - JSON*/
export const getDataSourcesResponse = {
  savedObjects: [
    {
      id: 'test',
      type: 'data-source',
      description: 'test datasource',
      title: 'test',
      get(field: string) {
        const me: any = this || {};
        return me[field];
      },
    },
    {
      id: 'test2',
      type: 'data-source',
      description: 'test datasource2',
      title: 'test',
      get(field: string) {
        const me: any = this || {};
        return me[field];
      },
    },
    {
      id: 'alpha-test',
      type: 'data-source',
      description: 'alpha test datasource',
      title: 'alpha-test',
      get(field: string) {
        const me: any = this || {};
        return me[field];
      },
    },
    {
      id: 'beta-test',
      type: 'data-source',
      description: 'beta test datasource',
      title: 'beta-test',
      get(field: string) {
        const me: any = this || {};
        return me[field];
      },
    },
  ],
};

export const existingDatasourceNamesList = [
  'test123',
  'testTest20',
  'TeSt',
  'duplicateTest',
  'dup20',
];

export const getMappedDataSources = [
  {
    id: 'test',
    description: 'test datasource',
    title: 'test',
    sort: 'test',
  },
  {
    id: 'test2',
    description: 'test datasource2',
    title: 'test',
    sort: 'test',
  },
  {
    id: 'alpha-test',
    description: 'alpha test datasource',
    title: 'alpha-test',
    sort: 'alpha-test',
  },
  {
    id: 'beta-test',
    description: 'beta test datasource',
    title: 'beta-test',
    sort: 'beta-test',
  },
];

export const mockDataSourceAttributesWithAuth = {
  id: 'test',
  title: 'create-test-ds',
  description: 'jest testing',
  endpoint: 'https://test.com',
  auth: {
    type: AuthType.UsernamePasswordType,
    credentials: {
      username: 'test123',
      password: 'test123',
    },
  },
};

export const mockDataSourceAttributesWithNoAuth = {
  id: 'test123',
  title: 'create-test-ds123',
  description: 'jest testing',
  endpoint: 'https://test.com',
  auth: {
    type: AuthType.NoAuth,
    credentials: undefined,
  },
};
export const getDataSourceByIdWithCredential = {
  attributes: {
    id: 'alpha-test',
    title: 'alpha-test',
    endpoint: 'https://test.com',
    description: 'alpha test data source',
    auth: {
      type: AuthType.UsernamePasswordType,
      credentials: {
        username: 'test123',
      },
    },
  },
};

export const getDataSourceByIdWithoutCredential = {
  attributes: {
    ...getDataSourceByIdWithCredential.attributes,
    auth: {
      type: AuthType.NoAuth,
      credentials: undefined,
    },
  },
  references: [],
};

export const mockResponseForSavedObjectsCalls = (
  savedObjectsClient: SavedObjectsClientContract,
  savedObjectsMethodName: 'get' | 'find' | 'create' | 'delete' | 'update',
  response: any
) => {
  (savedObjectsClient[savedObjectsMethodName] as jest.Mock).mockResolvedValue(response);
};

export const mockErrorResponseForSavedObjectsCalls = (
  savedObjectsClient: SavedObjectsClientContract,
  savedObjectsMethodName: 'get' | 'find' | 'create' | 'delete' | 'update'
) => {
  (savedObjectsClient[savedObjectsMethodName] as jest.Mock).mockRejectedValue(
    throwError(new Error('Error while fetching data sources'))
  );
};
