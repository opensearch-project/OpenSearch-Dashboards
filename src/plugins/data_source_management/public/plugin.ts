/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { CoreSetup, CoreStart, Plugin } from '../../../core/public';

import { PLUGIN_NAME } from '../common';

import { ManagementSetup } from '../../management/public';
import { IndexPatternManagementSetup } from '../../index_pattern_management/public';
import { DataSourceColumn } from './components/data_source_column/data_source_column';

export interface DataSourceManagementSetupDependencies {
  management: ManagementSetup;
  indexPatternManagement: IndexPatternManagementSetup;
}

const DSM_APP_ID = 'dataSources';

export class DataSourceManagementPlugin
  implements Plugin<void, void, DataSourceManagementSetupDependencies> {
  public setup(
    core: CoreSetup,
    { management, indexPatternManagement }: DataSourceManagementSetupDependencies
  ) {
    const opensearchDashboardsSection = management.sections.section.opensearchDashboards;

    if (!opensearchDashboardsSection) {
      throw new Error('`opensearchDashboards` management section not found.');
    }

    const savedObjectPromise = core
      .getStartServices()
      .then(([coreStart]) => coreStart.savedObjects);
    const httpPromise = core.getStartServices().then(([coreStart]) => coreStart.http);
    const column = new DataSourceColumn(savedObjectPromise, httpPromise);
    indexPatternManagement.columns.register(column);

    opensearchDashboardsSection.registerApp({
      id: DSM_APP_ID,
      title: PLUGIN_NAME,
      order: 1,
      mount: async (params) => {
        const { mountManagementSection } = await import('./management_app');

        return mountManagementSection(core.getStartServices, params);
      },
    });
  }

  public start(core: CoreStart) {}

  public stop() {}
}
