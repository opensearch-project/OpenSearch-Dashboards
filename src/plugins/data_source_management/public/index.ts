/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { DataSourceManagementPlugin } from './plugin';

// This exports static code and TypeScript types,
// as well as, OpenSearch Dashboards Platform `plugin()` initializer.
export function plugin() {
  return new DataSourceManagementPlugin();
}
export { DataSourceManagementPluginStart } from './types';
