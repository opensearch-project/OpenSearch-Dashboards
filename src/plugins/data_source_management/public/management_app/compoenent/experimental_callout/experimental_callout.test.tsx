/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { mount } from 'enzyme';
import { ExperimentalCallout } from './experimental_callout';
import { DocLinksStart } from 'opensearch-dashboards/public';
import { docLinks } from '../../../mocks';

const titleIdentifier = '.euiCallOutHeader__title';
const descriptionIdentifier = '[data-test-subj="data-source-experimental-call"]';
const expectedTitleText = 'Experimental Feature';
const expectedDescriptionText =
  'Experimental FeatureThe feature is experimental and should not be used in a production environment. Any index patterns, visualization, and observability panels will be impacted if the feature is deactivated. For more information see Data Source Documentation(opens in a new tab or window) To leave feedback, visit OpenSearch Forum';

describe('Datasource experimental callout component', () => {
  test('should render normally', () => {
    const mockedDocLinks = docLinks as DocLinksStart;
    const component = mount(<ExperimentalCallout docLinks={mockedDocLinks} />);
    const titleText = component.find(titleIdentifier).text();
    const descriptionText = component.find(descriptionIdentifier).last().text();

    expect(titleText).toBe(expectedTitleText);
    expect(descriptionText).toBe(expectedDescriptionText);
    expect(component).toMatchSnapshot();
  });
});
