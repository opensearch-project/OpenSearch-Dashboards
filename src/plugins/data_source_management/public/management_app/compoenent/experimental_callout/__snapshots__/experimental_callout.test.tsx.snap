// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datasource experimental callout component should render normally 1`] = `
<ExperimentalCallout
  docLinks={
    Object {
      "links": Object {
        "noDocumentation": Object {
          "indexPatterns": Object {
            "introduction": "",
          },
          "scriptedFields": Object {},
        },
      },
    }
  }
>
  <EuiCallOut
    data-test-subj="data-source-experimental-call"
    iconType="iInCircle"
    title="Experimental Feature"
  >
    <div
      className="euiCallOut euiCallOut--primary"
      data-test-subj="data-source-experimental-call"
    >
      <div
        className="euiCallOutHeader"
      >
        <EuiIcon
          aria-hidden="true"
          className="euiCallOutHeader__icon"
          color="inherit"
          size="m"
          type="iInCircle"
        >
          <span
            aria-hidden="true"
            className="euiCallOutHeader__icon"
            color="inherit"
            data-euiicon-type="iInCircle"
            size="m"
          />
        </EuiIcon>
        <span
          className="euiCallOutHeader__title"
        >
          Experimental Feature
        </span>
      </div>
      <EuiText
        color="default"
        size="s"
      >
        <div
          className="euiText euiText--small"
        >
          <EuiTextColor
            color="default"
            component="div"
          >
            <div
              className="euiTextColor euiTextColor--default"
            >
              <p>
                The feature is experimental and should not be used in a production environment. Any index patterns, visualization, and observability panels will be impacted if the feature is deactivated. For more information see 
                <EuiLink
                  href=""
                  target="_blank"
                >
                  <a
                    className="euiLink euiLink--primary"
                    href=""
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Data Source Documentation
                    <EuiIcon
                      aria-label="External link"
                      className="euiLink__externalIcon"
                      size="s"
                      type="popout"
                    >
                      <span
                        aria-label="External link"
                        className="euiLink__externalIcon"
                        data-euiicon-type="popout"
                        size="s"
                      />
                    </EuiIcon>
                    <EuiScreenReaderOnly>
                      <span
                        className="euiScreenReaderOnly"
                      >
                        <EuiI18n
                          default="(opens in a new tab or window)"
                          token="euiLink.newTarget.screenReaderOnlyText"
                        >
                          (opens in a new tab or window)
                        </EuiI18n>
                      </span>
                    </EuiScreenReaderOnly>
                  </a>
                </EuiLink>
                 
                To leave feedback, visit 
                <EuiLink
                  target="_blank"
                >
                  <button
                    className="euiLink euiLink--primary"
                    disabled={false}
                    type="button"
                  >
                    OpenSearch Forum
                  </button>
                </EuiLink>
              </p>
            </div>
          </EuiTextColor>
        </div>
      </EuiText>
    </div>
  </EuiCallOut>
  <EuiSpacer
    size="m"
  >
    <div
      className="euiSpacer euiSpacer--m"
    />
  </EuiSpacer>
</ExperimentalCallout>
`;
